packages/api/src/controllers/sensorController.js

```diff
@@ -0,0 +1,319 @@
+          parseFunction: (val) => val.trim(),
+          validator: (val) => val.length <= 100,
+          required: false,
+        },
+        Model: {
+          key: 'model',
+          parseFunction: (val) => val.trim(),
+          validator: (val) => val.length <= 100,
+          required: false,
+        },
+        Hardware_version: {
+          key: 'hardware_version',
+          parseFunction: (val) => val.trim(),
+          validator: (val) => val.length <= 100,
+          required: false,
+        },
+      });
+      if (errors.length > 0) {
+        res.status(400).send({ error_type: 'validation_failure', errors });
+      } else {
+        console.log(data);
+        const organization = await createOrganization(farm_id, access_token);
+        console.log(organization);
+        const esids = data.reduce((previous, current) => {
+          if (current.brand === 'Ensemble Scientific' && current.external_id) {
+            previous.push(current.external_id);
+          }
+          return previous;
+        }, []);
+        await registerOrganizationWebhook(farm_id, organization.organization_uuid, access_token);
+        const registeredSensors = await bulkSensorClaim(
+          access_token,
+          organization.organization_uuid,
+          esids,
+        );
+        console.log(registeredSensors);
+        for (const sensor of data) {
+          if (registeredSensors.success.includes(sensor.external_id)) {
+            const savedSensor = await sensorModel.query().insert({
+              farm_id,
+              name: sensor.name,
+              grid_points: {
+                lat: sensor.latitude,
+                lng: sensor.longitude,
+              },
+              partner_id: 1,
+              depth: sensor.depth,
+              external_id: sensor.external_id,
+            });
+            const readingTypes = await Promise.all(
+              sensor.reading_types.map((r) => {
+                return PartnerReadingTypeModel.getReadingTypeByReadableValue(r);
+              }),
+            );
+            console.log(readingTypes);
+            await SensorReadingTypeModel.query().insert(
+              readingTypes.map((readingType) => {
+                return {
+                  partner_reading_type_id: readingType.partner_reading_type_id,
+                  sensor_id: savedSensor.sensor_id,
+                };
+              }),
+            );
+          }
+        }
+        if (
+          registeredSensors.success.length + registeredSensors.already_owned.length <
+          esids.length
+        ) {
+          res.status(400).send({
+            error_type: 'unable_to_claim_all_sensors',
+            registeredSensors,
+          });
+        } else {
+          res.status(200).send({ message: 'Successfully uploaded!' });
+        }
+      }
+    } catch (e) {
+      console.log(e);
+      res.status(500).send(e.message);
+    }
+  },
+
+  deleteSensor() {
+    return async (req, res) => {
```

Comment:
Please avoid functions that always and only return the same function. That's one too many functions.

Reply:
My bad, thanks for clarifying Calum